R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R

```diff
@@ -43,73 +46,138 @@ module_aglu_LB151.ag_MIRCA_ctry_C_GLU_irr_DISABLED <- function(command, ...) {
+    ag_HA_ha_ctry_crop %>%
+      select(-irrHA) ->                                                         # Remove extra columns
+      L151.ag_rfdHA_ha_ctry_crop
+
+    ag_HA_ha_ctry_crop %>%
+      select(-rfdHA) ->                                                         # Remove extra columns
+      L151.ag_irrHA_ha_ctry_crop
+
+    # Calculate production on irrigated and rainfed land, by country, GTAP crop, and GLU
+    # Use the ratio of irrigated to rainfed yield from the FAO CROSIT database,
+    # in combination with the harvested area data computed above to separate irrigated
+    # and rainfed production.
+    # First, prepare the country mapping file. The old data system uses the first entry when duplicates
+    # are found (which happens a lot in this file).
+    AGLU_ctry %>%
+      select(iso, CROSIT_country_ID) %>%
+      distinct(iso, .keep_all=TRUE) ->
+      Unique_AGLU_ctry
+
+    # Next, Calculate the yield ratio for each country and crop, based on the CROSIT database
+    L100.LDS_ag_HA_ha %>%
+      select(iso, GLU, GTAP_crop) %>%
+      left_join(Unique_AGLU_ctry[ c( "iso", "CROSIT_country_ID") ], by = "iso" ) %>%                                    # Map in CROSIT country ids
+      left_join(FAO_ag_items_PRODSTAT[ c("GTAP_crop","CROSIT_cropID")], by = "GTAP_crop" ) %>%                          # Map in CROSIT crop ids
+      left_join(FAO_ag_CROSIT[ c( "country_ID", "crop_ID", "Yield_kgHa_irrigated", "Yield_kgHa_rainfed", "year" )],     # Map in CROSIT irrigated and rainfed yields
+                 by = c( "CROSIT_country_ID" = "country_ID", "CROSIT_cropID" = "crop_ID")) %>%
+
+      # CROSIT data includes historic and future years, we only want the historic year. Additionally, CROSIT data is missing some crops
+      # we don't want to remove those. We'll set their yield ratio to 1 later.
+      filter(year == CROSIT_HISTORICAL_YEAR | is.na(year)) %>%
+      mutate(yieldratio = Yield_kgHa_irrigated / Yield_kgHa_rainfed) %>%                                                # Compute yield ratio
+      select(-Yield_kgHa_irrigated, -Yield_kgHa_rainfed) %>%
+      mutate(yieldratio=if_else( is.na(yieldratio), 1, yieldratio )) %>%                                                # Replace NAs with 1
+      mutate(yieldratio=if_else( yieldratio==0, 1, yieldratio )) %>%                                                    # Replace zeros with 1
+      mutate(yieldratio=if_else( is.infinite(yieldratio), 1, yieldratio ))  ->                                          # Replace Infs with 1
+      Yieldratio_ctry_crop
+
+    # Now, use the yield ratio to solve for the production shares (irrigated versus rainfed)
+    # This calculation preserves the harvested area shares and yield ratios computed above.
+    ag_HA_ha_ctry_crop %>%
+      left_join( Yieldratio_ctry_crop[ c( "iso", "GLU", "GTAP_crop", "yieldratio") ], by = c( "iso", "GLU", "GTAP_crop") ) %>%  # Map in the yield ratio
```

Comment:
For consistent style might do `left_join(select(Yieldratio_ctry_crop, iso, ...), by = ...)`     ...not critical though